src/components/SubmitTokenInfo/index.tsx

```diff
@@ -0,0 +1,250 @@
+import styles from './styles.module.scss'
+import { ImgUpload } from './ImgUpload'
+import CommonButton from '../CommonButton'
+import CommonModal from '../CommonModal'
+import { submitTokenInfo } from '../../services/ExplorerService/fetcher'
+import CommonSelect from '../CommonSelect'
+import { isMainnet } from '../../utils/chain'
+import { MainnetContractHashTags, TestnetContractHashTags } from '../../constants/scripts'
+
+type Props = {
+  isOpen: boolean
+  onClose: () => void
+}
+
+const LabelTooltip = ({ title, icon }: { title: string; icon?: string }) => (
+  <Tooltip placement="bottom" title={title}>
+    <img src={icon ?? HelpIcon} alt="tooltip" className={styles.tooltipIcon} />
+  </Tooltip>
+)
+
+export const SubmitTokenInfo = (prop: Props) => {
+  const { t } = useTranslation()
+  const { onClose, isOpen } = prop
+  const [args, setArgs] = useState('')
+
+  const [symbol, setSymbol] = useState('')
+  const [name, setName] = useState('')
+  const [decimal, setDecimal] = useState('')
+  const [description, setDescription] = useState('')
+  const [website, setWebsite] = useState('')
+  const [creatorEmail, setCreatorEmail] = useState('')
+  const [logo, setLogo] = useState<string | null>(null)
+
+  const scriptDataList = isMainnet() ? MainnetContractHashTags : TestnetContractHashTags
+  const tokenTypeOptions = scriptDataList
+    .filter(scriptData => scriptData.tag.includes('sudt'))
+    .reverse() // sUDT should come first, then sUDT(deprecated)
+    .map(scriptData => ({
+      value: scriptData.codeHashes[0],
+      label: scriptData.tag,
+    }))
+  const [codeHash, setCodeHash] = useState<string>(tokenTypeOptions[0].value)
+  const handleTokenTypesChange = (value: string) => {
+    setCodeHash(value)
+  }
+
+  useEffect(() => {
+    // scroll to top when open modal
+    if (isOpen) {
+      window.scrollTo(0, 0)
+    }
+  }, [isOpen])
+
+  const handleImgChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files?.[0]
+    if (file) {
+      const reader = new FileReader()
+      let baseString: string | null = null
+      reader.onloadend = function () {
+        baseString = reader.result as string
+        setLogo(baseString)
+      }
+      reader.readAsDataURL(file)
+    }
+  }
+
+  const handleClose = () => {
+    onClose()
+    setArgs('')
+    setSymbol('')
+    setName('')
+    setDecimal('')
+    setDescription('')
+    setWebsite('')
+    setCreatorEmail('')
+    setLogo(null)
+    onClose()
+  }
+  const parsedDecimal = parseInt(decimal, 10)
+
+  const validateFields = () => {
+    if (Number.isNaN(parsedDecimal)) {
+      return false
+    }
+    if (!codeHash) {
+      return false
+    }
+    if (!symbol) {
+      return false
+    }
+    if (!args) {
+      return false
+    }
+    if (!website) {
+      return false
+    }
+    if (!creatorEmail) {
+      return false
+    }
+    return true
+  }
+
+  const handleConfirm = async () => {
+    if (!validateFields()) {
```

Comment:
This function could be refactored into an expression directly, e.g. `const isValid = !!codeHash && !!symbol`